packages/manager/apps/pci-load-balancer/src/components/detail/listeners/ListenerForm.page.tsx

```diff
@@ -0,0 +1,320 @@
+import { useTranslation } from 'react-i18next';
+import {
+  LISTENER_POOL_PROTOCOL_COMBINATION,
+  PROTOCOLS,
+  PROTOCOLS_PORT,
+} from '@/constants';
+import {
+  TLoadBalancerListener,
+  TLoadBalancerPool,
+} from '@/api/data/load-balancer';
+import Popover from './Popover';
+
+type TFormState = {
+  name: string;
+  protocol: string;
+  port: number;
+  pool: TLoadBalancerPool | null;
+};
+
+export type ListenerFormProps = {
+  listener?: TLoadBalancerListener | null;
+  pools: TLoadBalancerPool[];
+  isPending: boolean;
+  isEditing?: boolean;
+  onSubmit: (state: TFormState) => void;
+  onCancel: () => void;
+};
+
+export default function ListenerForm({
+  listener = null,
+  isEditing = false,
+  pools,
+  isPending,
+  onCancel,
+  onSubmit,
+}: ListenerFormProps) {
+  const { t: tCommon } = useTranslation('pci-common');
+  const { t } = useTranslation('octavia-load-balancer-listeners-create');
+  const { t: tListeners } = useTranslation('octavia-load-balancer-listeners');
+
+  const PORT_MIN_VALUE = 1;
+  const PORT_MAX_VALUE = 65535;
+
+  const [formState, setFormState] = useState<TFormState>({
+    name: '',
+    protocol: '',
+    port: 1,
+    pool: null,
+  });
+
+  const [filteredPools, setFilteredPools] = useState<TLoadBalancerPool[]>([]);
+  const [isTouched, setIsTouched] = useState({ name: false, port: false });
+
+  const getFilteredPools = (): TLoadBalancerPool[] =>
+    pools.filter(
+      ({ listenerId, protocol }) =>
+        !listenerId &&
+        LISTENER_POOL_PROTOCOL_COMBINATION[formState.protocol]?.includes(
+          protocol,
+        ),
+    ) || [];
+
+  useEffect(() => {
+    if (isEditing && listener && pools?.length) {
+      setFormState({
+        name: listener.name,
+        protocol: listener.protocol,
+        port: listener.port,
+        pool: pools.find((pool) => pool.id === listener.defaultPoolId) ?? null,
+      });
+    }
+  }, [listener, isEditing, pools]);
+
+  useEffect(() => {
+    if (pools) {
+      setFilteredPools(formState.protocol ? getFilteredPools() : pools);
+    }
+  }, [pools, formState.protocol]);
+
+  const handleProtocolChange = (event) => {
+    const selectedProtocol = event.detail.value;
+
+    setFormState((prev) => ({
+      ...prev,
+      protocol: selectedProtocol,
+      port: PROTOCOLS_PORT[selectedProtocol] || 1,
+    }));
+
+    if (
+      (formState.pool &&
+        !filteredPools.find(({ id }) => id === formState.pool.id)) ||
+      selectedProtocol === PROTOCOLS.PROMETHEUS
+    ) {
+      setFormState((prev) => ({
+        ...prev,
+        pool: null,
+      }));
+    }
+  };
+
+  const handleChange = (
+    key: string,
+    value: string | number | TLoadBalancerPool,
+  ) => {
+    setFormState((prev) => ({
+      ...prev,
+      [key]: value,
+    }));
+  };
+
+  const handleBlur = (key: 'name' | 'port') => {
+    setIsTouched((prev) => ({
+      ...prev,
+      [key]: true,
+    }));
+  };
+
+  const portError = () => {
```

Comment:
is a memo, this way it will be executed once per render

Reply:
Fixed